diff --git a/test/dma_buf.cpp b/test/dma_buf.cpp
index a34bc44..95718a1 100644
--- a/test/dma_buf.cpp
+++ b/test/dma_buf.cpp
@@ -4,6 +4,7 @@
 #include <limits>
 #include <variant>
 #include <cerrno>
+#include <cstdint>
 
 #include <sys/ioctl.h>
 #include <sys/mman.h>
diff --git a/test/lock.cpp b/test/lock.cpp
index 941006e..0272126 100644
--- a/test/lock.cpp
+++ b/test/lock.cpp
@@ -3,6 +3,7 @@
 
 #include <optional>
 #include <string>
+#include <cstdint>
 
 #include <errno.h>
 #include <sys/ioctl.h>
