diff --git a/device/cpuset_lib.cpp b/device/cpuset_lib.cpp
index f8d5e03..93e89b7 100644
--- a/device/cpuset_lib.cpp
+++ b/device/cpuset_lib.cpp
@@ -9,6 +9,7 @@
 #include <thread>
 #include "device/device_api.h"
 #include <filesystem>
+#include <fmt/std.h>
 namespace tt {
 
 namespace fs = std::filesystem;
diff --git a/device/tt_silicon_driver.cpp b/device/tt_silicon_driver.cpp
index fb243ed..c8950ae 100644
--- a/device/tt_silicon_driver.cpp
+++ b/device/tt_silicon_driver.cpp
@@ -4342,7 +4342,7 @@ void tt_SiliconDevice::set_power_state(tt_DevicePowerState device_state) {
             set_pcie_power_state(device_state);
         } else {
             int exit_code = set_remote_power_state(chip, device_state);
-            log_assert(exit_code == 0, "Failed to set power state to {} with exit code: {}", device_state, exit_code);
+            log_assert(exit_code == 0, "Failed to set power state with exit code: {}", exit_code);
         }
     }
 }
